ui/webui/src/components/storage/StorageConfiguration.jsx

```diff
@@ -253,15 +251,49 @@ const GuidedPartitioning = ({ deviceData, selectedDisks, idPrefix, scenarios, st
     ));
     const [isDetailExpanded, setIsDetailExpanded] = useState(false);
     const [detailContent, setDetailContent] = useState("");
+    const [requiredSize, setRequiredSize] = useState();
+    const [diskTotalSpace, setDiskTotalSpace] = useState();
+    const [diskFreeSpace, setDiskFreeSpace] = useState();
+    const [hasPartitions, setHasPartitions] = useState();
+
+    useEffect(() => {
+        const updateSizes = async () => {
+            const diskTotalSpace = await getDiskTotalSpace({ diskNames: selectedDisks }).catch(console.error);
+            const diskFreeSpace = await getDiskFreeSpace({ diskNames: selectedDisks }).catch(console.error);
+
+            setDiskTotalSpace(diskTotalSpace);
+            setDiskFreeSpace(diskFreeSpace);
+        };
+        updateSizes();
+    }, [selectedDisks]);
+
+    useEffect(() => {
+        const updateRequiredSize = async () => {
+            const requiredSpace = await getRequiredSpace().catch(console.error);
+            const requiredSize = await getRequiredDeviceSize({ requiredSpace }).catch(console.error);
+
+            setRequiredSize(requiredSize);
+        };
+        updateRequiredSize();
+    }, []);
+
+    useEffect(() => {
+        const hasPartitions = selectedDisks.some(device => deviceData[device]?.children.v.some(child => deviceData[child]?.type.v === "partition"));
+
+        setHasPartitions(hasPartitions);
+    }, [selectedDisks, deviceData]);
 
     useEffect(() => {
         const updateScenarioState = async (scenarios) => {
```

Comment:
Maybe all the hook content can become sync now?

Reply:
Yes you 're right! thanks for noticing!